src/__tests__/parseWithPointers.ts

```diff
@@ -18,4 +20,111 @@ describe('json parser', () => {
 }`)
     ).toMatchSnapshot();
   });
+
+  test('parse complex', () => {
+    expect(
+      parseWithPointers(`{
+  "users": [
+    {
+      "name": "Markku",
+      "address": {
+         "country": "Finland",
+         "city": "Helsinki"
+       },
+      "age": 17,
+      "adult": false
+    },
+    {
+      "name": "Dennis",
+      "country": "Denmark",
+      "age": 18,
+      "adult": true
+    }
+  ]
+}`)
+    ).toMatchSnapshot();
+  });
+
+  test('does not allow comments by default', () => {
+    expect(
+      parseWithPointers(`{
+      // a comment
+      "name": "Antti",
+      /* a
+       very
+       block 
+       comment  
+       */
+      "city": "Vantaa"
+    }`)
+    ).toHaveProperty('validations', [
+      expect.objectContaining({
```

Comment:
Cool, I did not know it was possible to nest multiple assertions in `toHaveProperty` call.

Reply:
Yup, most of matchers accept custom matchers.